src/xsdba/_adjustment.py

```diff
@@ -145,18 +145,28 @@ def dqm_train(
     `jitter_over_thresh_value` and `jitter_over_thresh_upper_bnd` must be both be specified to
     use `jitter_over_thresh`, or both be None (default) to skip it.
     """
+    # Ensure we only reduce on valid dims, allows for extra dims like "realization" on the sim
+    ref_dim = Grouper.filter_dim(ds.ref, dim)
+    sim_dim = Grouper.filter_dim(ds.hist, dim)
+
+    # this might be empty, in which case this does nothing
+    extra_dim = list(set(sim_dim) - set(ref_dim))
+    ref = ds.ref.expand_dims({d: [0] for d in extra_dim})
+    hist = ds.hist
+
+    # This ensures we can merge ref and hist even if they have
+    # different sizes for extra_dim
+    ref = ref.assign_coords({d: ref[d] for d in ref.dims})
+    hist = hist.assign_coords({d: hist[d] for d in hist.dims})
```

Comment:
What's the error if you have dims without coords ? Instead of adding fake coords, we could simply fail, asking the user to fix the data ?